crates/egui/src/widgets/image.rs

```diff
@@ -43,15 +44,93 @@ use epaint::{util::FloatOrd, RectShape};
 #[must_use = "You should put this widget in an ui with `ui.add(widget);`"]
 #[derive(Debug, Clone)]
 pub struct Image<'a> {
-    source: ImageSource<'a>,
+    sources: ImageSources<'a>,
     texture_options: TextureOptions,
     image_options: ImageOptions,
     sense: Sense,
     size: ImageSize,
     pub(crate) show_loading_spinner: Option<bool>,
 }
 
+#[derive(Debug, Clone)]
+/// This type tells the [`Ui`] if the image is an animated Image and when to render the next frame.
+///
+/// This is used by [`Image::new_animation`]
+pub enum ImageSources<'a> {
+    Multi(Vec<(ImageSource<'a>, Duration)>),
+    Single(ImageSource<'a>),
```

Comment:
these names might be more descriptive

```suggestion
    /// A static image.
    Static(ImageSource<'a>),
    
    /// An animation of images, with the length of time to show each frame.
    Animated(Vec<(ImageSource<'a>, Duration)>),
```